pkg/cloud/gcp/gcs_kms.go

```diff
@@ -74,22 +77,46 @@ func MakeGCSKMS(ctx context.Context, uri string, env cloud.KMSEnv) (cloud.KMS, e
 
 	switch kmsURIParams.auth {
 	case "", cloud.AuthParamSpecified:
-		if kmsURIParams.credentials == "" {
-			return nil, errors.Errorf(
-				"%s is set to '%s', but %s is not set",
-				cloud.AuthParam,
-				cloud.AuthParamSpecified,
-				CredentialsParam,
-			)
+		if temporaryTokenEnabled.Get(&env.ClusterSettings().SV) {
```

Comment:
ditto comment as below about restructuring the if condition

Reply:
done. feature flag is removed.